hazelcast/src/main/java/com/hazelcast/internal/util/ConcurrencyUtil.java

```diff
@@ -148,6 +148,31 @@ public static <K, V> V getOrPutSynchronized(ConcurrentMap<K, V> map, K key, Cont
         return value;
     }
 
+    /**
+     * @return the {@code value} corresponding to {@code key} in the {@code map}. If {@code key} is not mapped in {@link map},
+     *         then a {@code value} is computed using {@code func}, inserted into the map and returned.
+     *         <p>
```

Comment:
you described entire behavior under `@return`, would be better before

Reply:
211b18db722ef959d9fcde66998d64b6cc52c612